### PR TITLE
[#IR-331][#IR-330] As a dev, I have added support to delete the notes, highlights, goals and tickets.

### PR DESCRIPTION
**Pull request by chahat-arora**
also, I can see the activity logs on the addition and deletion of feedbacks and tasks.

### CODE REVIEW

**Review by sandipagarwal — APPROVED** [review]

File: apps/retrospective/services/sprint_task.go [hunk 1]
```diff
@@ -132,6 +132,48 @@ func (service SprintTaskService) Update(sprintTaskID string, retroID string, spr
...
+	gormDB := service.DB
+
+	var sprintTask retroModels.SprintTask
+	err := gormDB.Model(retroModels.SprintTask{}).Where("id = ?", sprintTaskID).Scan(&sprintTask).Error
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	var sprintTasksCount int
+	err = gormDB.Model(&retroModels.SprintTask{}).Where("task_id = ?", sprintTask.TaskID).Count(&sprintTasksCount).Error
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task.go]: Please add proper comments here!!

File: apps/retrospective/services/sprint_task.go [hunk 2]
```diff
@@ -132,6 +132,48 @@ func (service SprintTaskService) Update(sprintTaskID string, retroID string, spr
 	return service.Get(sprintTaskID, retroID, sprintID)
 }
 
+// Delete ...
+func (service SprintTaskService) Delete(sprintTaskID string, retroID string, sprintID string) (int, error) {
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task.go:136]: Use DB transactions here.

File: apps/retrospective/services/sprint_task.go [hunk 3]
```diff
@@ -132,6 +132,48 @@ func (service SprintTaskService) Update(sprintTaskID string, retroID string, spr
...
+		return http.StatusInternalServerError, err
+	}
+
+	// delete sprintTask.
+	err = gormDB.Where("id = ?", sprintTaskID).Delete(retroModels.SprintTask{}).Error
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+	// delete sprintMemberTask
+	err = gormDB.Where("sprint_task_id = ?", sprintTaskID).Delete(retroModels.SprintMemberTask{}).Error
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task.go]: Shouldn't we delete SMTs first as Sprint task is used in those SMTs?

**Review by himanshu-garg-165 — CHANGES_REQUESTED** [review]

File: apps/retrospective/services/sprint_task.go (hunk 3 above)
Comment (chahat-arora) [reply apps/retrospective/services/sprint_task.go]: done.

File: apps/retrospective/services/sprint_task.go (hunk 1 above)
Comment (chahat-arora) [reply apps/retrospective/services/sprint_task.go]: done.

File: apps/retrospective/services/sprint_task.go (hunk 2 above)
Comment (chahat-arora) [reply apps/retrospective/services/sprint_task.go:136]: done.

**Review by himanshu-garg-165 — APPROVED** [review]

**Pull request merged**